icuSources/common/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+file(GLOB_RECURSE _ICUCommonSources "*.cpp")
+add_library(ICUCommon STATIC ${_ICUCommonSources})
+
+target_include_directories(ICUCommon
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/unicode/>
+        include/)
+
+target_link_libraries(ICUCommon PRIVATE stdc++)
```

Comment:
This shouldn't be necessary. From chatting with @compnerd, we might want to enable C++ interop so that the linker-driver is `clang++` instead of just `clang` so that it implicitly links the C++ standard library.